### PR TITLE
Public API Documentation

### PR DESCRIPTION
**Pull request by philipuvarov**
Add documentation for public API

### CODE REVIEW

**Comment by philipuvarov** [comment]
covers this issue: https://github.com/avito-tech/Konveyor/issues/2

File: konveyor/src/main/java/com/avito/konveyor/blueprint/ItemView.kt [hunk 1]
```diff
@@ -1,5 +1,19 @@
 package com.avito.konveyor.blueprint
 
+import android.support.v7.widget.RecyclerView
+
+/**
+ * Marker interface that represents an
+ * abstractions over [RecyclerView.ViewHolder]
```
Comment (dmitriigriazin) [inline konveyor/src/main/java/com/avito/konveyor/blueprint/ItemView.kt]: `an abstractions` grammar nazi

File: konveyor/src/main/java/com/avito/konveyor/adapter/AdapterPresenter.kt [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
+     * Will be called on [RecyclerView.Adapter.getItemId]
+     */
     fun getItemId(position: Int): Long
 
+    /**
+     * Method to notify [AdapterPresenter] that data has been changed
+     *
+     * <em>Note:</em> if you are using [SimpleRecyclerAdapter] you still need to call
+     * [RecyclerView.Adapter.notifyDataSetChanged] in order to notify your adapter about changes
+     */
```
Comment (dmitriigriazin) [inline konveyor/src/main/java/com/avito/konveyor/adapter/AdapterPresenter.kt:38]: People would be interested, where all the other recycler `notifyItem...()` calls are.

File: konveyor/src/main/java/com/avito/konveyor/validation/ValidationPolicy.kt [hunk 3]
```diff
@@ -1,6 +1,15 @@
 package com.avito.konveyor.validation
 
+import com.avito.konveyor.ItemBinder
+import com.avito.konveyor.konveyorConfiguration
 
+/**
+ * If set to `true` it will make [ItemBinder] throw in case of collisions or unregistered data
```
Comment (dmitriigriazin) [inline konveyor/src/main/java/com/avito/konveyor/validation/ValidationPolicy.kt:7]: No javadoc for ItemBinder? Even if we make it internal (?) it'd help to understand the abstractions. There is no info about ItemBinder in README either, though you're using it in code snippet

Comment (philipuvarov) [reply konveyor/src/main/java/com/avito/konveyor/validation/ValidationPolicy.kt:7]: Yeah, you are right, I will add the docs for ItemBinder

File: konveyor/src/main/java/com/avito/konveyor/adapter/AdapterPresenter.kt (hunk 2 above)
Comment (philipuvarov) [reply konveyor/src/main/java/com/avito/konveyor/adapter/AdapterPresenter.kt:38]: Will be shown in the sample project

**Comment by philipuvarov** [comment]
@bagza please review new commit with discussed improvements

**Review by dmitriigriazin — APPROVED** [review]

**Pull request merged**